lib/ws-channel.js

```diff
@@ -56,43 +59,68 @@ WebsocketChannel.prototype._onNotification = function(message) {
 function nullCallback() {
 }
 
-WebsocketChannel.prototype.connect = function(uri, _token) {
-  var token = _token || url.parse(uri).auth;
+WebsocketChannel.prototype.connect = function(uri, options) {
+  assert(!this._clientToken);
+  assert(!this._sessionToken);
+  
+  options = options || {};
+  this._clientToken = options.clientToken || url.parse(uri).auth;
+  this._sessionToken = options.sessionToken || this._generateToken();
```

Comment:
i'm OK with client generating sessionId, but from our conversation, it sounded like if the server generated it on first connection and passed it back, it would simplify the client's tracking of whether it had ever sucesfully handshaked with the server. I see the isNew is based on sequence numbers below, which may be equivalent, but without a detailed reading of the sequence number logic, I don't know.